test/Microsoft.ML.AutoML.Tests/AutoFitTests.cs

```diff
@@ -401,21 +435,25 @@ public void AutoFitWithPresplittedData()
             var dataCV = context.Data.CrossValidationSplit(dataFull, numberOfFolds: 2);
 
             var modelFull = context.Auto()
-                .CreateBinaryClassificationExperiment(0)
+                .CreateBinaryClassificationExperiment(10)
                 .Execute(dataFull,
                     new ColumnInformation() { LabelColumnName = DatasetUtil.UciAdultLabel })
                 .BestRun
                 .Model;
 
+            // AutoMLExperiment can't run on canceled context.
+            // Therefore, we need to create a new context after an experiment is completed.
+            context = new MLContext(1);
             var modelTrainTest = context.Auto()
-                .CreateBinaryClassificationExperiment(0)
+                .CreateBinaryClassificationExperiment(10)
```

Comment:
I think we should change this behavior back to the old behavior. I don't think cancelling all running trials is a good idea. If I don't know how long my data will take to train and I set it for 2 hours and nothing finished by then, I would probably be pretty annoyed to have it cancel part of the way through.

Or maybe what would be better would be to have both ways possible. 1 way to force cancel everything, and 1 way to cancel but let it finish gracefully. Until we are able to have that though, I don't think we should forcefully cancel all running pipelines especially if non have finished yet.

Reply:
I would still vote for forcing canceling trial because the only situation for force canceling running trial ruins user experience is when there's no trial completed. Once there's a complete trial, we can simply kill all running trials and return that completed trial so the user can still get a result. Moreover, that only situation can also be avoided by making the first trial end quickly, via start from small model and start from a portion of dataset. ModelBuilder takes those techniques and it goes pretty well. The new AutoML also start from small model, and I'm working on a PR for starting from a portion of dataset when necessary.

Gracefully canceling, however, has more annoying situations. For example, when times up and AutoML is running a large model, it might take a few extra minutes if we want to wait for the final trial to be completed.

Reply:
We can probably do a mixed way: force canceling when there's completed trial, and wait for the first trial to complete if there's no completed trials.